src/convertToObject.js

```diff
@@ -6,7 +6,22 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const declarations = sourceString.split(';');
+
+  declarations.forEach((declaration) => {
+    const trimDeclaration = declaration.trim();
+
+    if (trimDeclaration) {
+      const [key, value] = trimDeclaration.split(':');
+
+      if (key && value) {
+        result[key.trim()] = value.trim();
+      }
+    }
+  });
```

Comment:
The 'forEach' loop can be replaced with a more functional approach using 'map' and 'filter', followed by 'Object.fromEntries' to create the object. This would adhere to the checklist item regarding avoiding loops and using iteration methods.